helm/postgres/templates/podmonitor.yaml

```diff
@@ -0,0 +1,46 @@
+{{- if .Values.monitoring }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: postgresql-exporter
+spec:
+  selector:
+    matchLabels:
+      postgres-operator.crunchydata.com/crunchy-postgres-exporter: "true"
+  podMetricsEndpoints:
+  - port: exporter
+    relabelings:
+    - sourceLabels: [__meta_kubernetes_namespace]
+      action: replace
+      targetLabel: kubernetes_namespace
+    - sourceLabels: [__meta_kubernetes_pod_name]
+      targetLabel: pod
+    - sourceLabels: [__meta_kubernetes_pod_label_postgres_operator_crunchydata_com_cluster,__meta_kubernetes_pod_label_pg_cluster]
+      targetLabel: cluster
+      separator: ""
+      replacement: '$1'
+    - sourceLabels: [__meta_kubernetes_namespace,cluster]
+      targetLabel: pg_cluster
+      separator: ":"
+      replacement: '$1$2'
+    - sourceLabels: [__meta_kubernetes_pod_ip]
+      targetLabel: ip
+      replacement: '$1'
+    - sourceLabels: [__meta_kubernetes_pod_label_postgres_operator_crunchydata_com_instance,__meta_kubernetes_pod_label_deployment_name]
+      targetLabel: deployment
+      replacement: '$1'
+      separator: ""
+    - sourceLabels: [__meta_kubernetes_pod_label_postgres_operator_crunchydata_com_role,__meta_kubernetes_pod_label_role]
+      targetLabel: role
+      replacement: '$1'
+      separator: ""
+    - sourceLabels: [dbname]
+      targetLabel: dbname
+      replacement: '$1'
+    - sourceLabels: [relname]
+      targetLabel: relname
+      replacement: '$1'
+    - sourceLabels: [schemaname]
+      targetLabel: schemaname
+      replacement: '$1'
```

Comment:
Could you explain why we need this relabeling?

Reply:
I simply copied the complete relabel_config from the original prometheus-config.
https://github.com/dgo19/postgres-operator-examples/blob/main/kustomize/monitoring/prometheus-config.yaml#L65-L73

There are several changes in the last 2 years ;-)

Reply:
> There are several changes in the last 2 years ;-)

Thanks for mentioning this ;-) I will also have a look directly to these files